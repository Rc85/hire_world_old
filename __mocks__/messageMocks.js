export const userMessage = {
    message_id: 1,
    belongs_to_job: 1,
    message_body: 'test',
    message_date: new Date(),
    message_type: 'User',
    message_sender: 'client',
    message_recipient: 'user',
    message_status: 'New',
    is_reply: 'false',
    message_modified_date: new Date()
}

export const systemUpdateMessage = Object.assign({}, userMessage);

systemUpdateMessage['message_type'] = 'Update';
systemUpdateMessage['message_sender'] = 'System';

export const systemConfirmMessage = Object.assign({}, userMessage);

systemConfirmMessage['message_type'] = 'Confirmation';
systemConfirmMessage['message_sender'] = 'System';