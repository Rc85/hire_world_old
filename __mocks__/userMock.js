export const User = {
    account_type: "Listing",
    allow_messaging: true,
    avatar_url: "/user_files/1/businessman-avatar-icon-01-.jpg",
    display_fullname: true,
    email_notifications: false,
    hide_email: false,
    user_address: "3475 E 27th Ave",
    user_bio: "Lorem ipsum dolor sit amet consectetur adipisicing elit. Assumenda ullam repellat maiores aperiam debitis iure, repudiandae maxime pariatur nam rerum quasi numquam aliquid illo temporibus amet officiis, veritatis ipsum? Illum. Lorem ipsum dolor sit amet consectetur adipisicing elit. Aspernatur similique quae tenetur repellat earum voluptate, reiciendis doloribus. Pariatur atque porro sunt autem quod harum soluta necessitatibus tempora hic. Molestias, harum! Lorem ipsum dolor sit amet consectetur adipisicing elit. Hic molestiae nihil, optio accusantium repellat ex cupiditate commodi similique facere minus maxime explicabo dolorum neque veniam corrupti ipsa debitis. Pariatur, necessitatibus. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ducimus voluptatem, a molestias aperiam obcaecati doloribus harum veritatis suscipit asperiores hic exercitationem molestiae aliquid soluta minima! Laborum placeat saepe dolores! Enim.",
    user_business_name: "RC Web Development",
    user_city: "Vancouver",
    user_city_code: "V5R1P8",
    user_country: "Canada",
    user_education: "BCIT",
    user_email: "rogerchin85@gmail.com",
    user_facebook: "https://www.facebook.com",
    user_firstname: "Roger",
    user_github: "https://github.com/Rc85",
    user_instagram: "https://www.instagram.com",
    user_last_login: "2018-12-10T00:32:16.540Z",
    user_lastname: "Chin",
    user_linkedin: "https://www.linkedin.com",
    user_phone: "604-889-8353",
    user_profile_id: 1,
    user_region: "British Columbia",
    user_setting_id: 1,
    user_title: "Web Developer",
    user_twitter: "https://www.twitter.com",
    user_website: "https://www.rogerchin.me",
    user_worldwide: null,
    username: "roger85"
}