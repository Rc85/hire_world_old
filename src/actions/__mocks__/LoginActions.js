export const User = {
    avatar_url: "../images/profile.png",
    business_name: "",
    default_location: false,
    display_business_name: false,
    display_contacts: false,
    display_fullname: false,
    email_notifications: true,
    hide_email: false,
    services_allowed: 3,
    user_address: null,
    user_bio: null,
    user_city: "",
    user_country: "",
    user_created_on: "2018-08-30T00:20:31.456Z",
    user_education: null,
    user_email: "hacker@hacker.com",
    user_facebook: null,
    user_firstname: "",
    user_github: null,
    user_id: 19,
    user_instagram: null,
    user_lastname: "",
    user_level: 0,
    user_linkedin: null,
    user_phone: null,
    user_region: "",
    user_status: "Active",
    user_title: null,
    user_twitter: null,
    user_type: null,
    user_website: null,
    user_worldwide: false,
    username: "hacker",
    view_counter: 0
}