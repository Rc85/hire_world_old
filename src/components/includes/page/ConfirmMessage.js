import React, { Component } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

class ConfirmMessage extends Component {
    constructor(props) {
        super(props);
        
        this.state = {
            decline: false,
            declineMessage: ''
        }
    }
    
    render() {
        let declineMessageInput, messageStatus;
        let alertClass = this.props.type;

        let buttons = <div className='mb-3'>
            <button id='confirm-message-approve-button' className='btn btn-success mr-1' onClick={() => this.props.approve()}>Approve</button>
            <button id='confirm-message-decline-button' className='btn btn-danger' onClick={() => {
                if (this.props.prompt) {
                    this.setState({decline: true});
                } else {
                    this.props.decline(this.state.declineMessage);
                }
            }}>Decline</button>
        </div>;

        if (this.props.user.username === this.props.message.message_recipient && this.props.message.message_status === 'New') {
            messageStatus = <h4><span className='badge badge-success'>New</span></h4>;
        }

        if (this.props.job.job_user_complete === false && this.props.job.job_client_complete === false) {
            buttons = <div id='confirm-message-declined-message' className='text-center'><strong><em>You declined the request.</em></strong></div>;
            alertClass = 'warning';
        } else if (this.props.job.job_user_complete && this.props.job.job_client_complete) {
            buttons = <div id='confirm-message-approved-message' className='text-center'><strong><em>You approved the request.</em></strong></div>;
            alertClass = 'success';
        }

        if (this.state.decline) {
            declineMessageInput = <div id='confirm-message-reason-input' className='mb-3'>
                <textarea rows='3' className='form-control w-100 mb-1' onChange={(e) => this.setState({declineMessage: e.target.value})} placeholder={`Briefly describe the reason why you are declining the other party's request.`}></textarea>
                <div className='text-right'>
                    <button className='btn btn-primary mr-1' onClick={() => {
                        this.props.decline(this.state.declineMessage)
                        this.setState({decline: false});
                    }}>Submit</button>
                    <button className='btn btn-secondary' onClick={() => this.setState({decline: false})}>Cancel</button>
                </div>
            </div>
        }

        return(
            <div className={`alert alert-${alertClass} mb-3 p-2`} role='alert'>
                <div className='d-flex-between-center'>
                    <small className='text-muted'>Sent {moment(this.props.message.message_date).from()} {this.props.message.message_modified_date ? <span>(Resent {moment(this.props.message.message_modified_date).fromNow()})</span> : ''}</small>
                    {messageStatus}
                </div>

                <div className='text-center mb-3 keep-format'><strong>{this.props.message.message_body}</strong></div>

                <hr/>

                {buttons}

                {declineMessageInput}

                <div className='text-right'><small className='text-muted'>Message ID: {this.props.message.message_id}</small></div>
            </div>
        )
    }
}

ConfirmMessage.propTypes = {
    message: PropTypes.object,
    job: PropTypes.object,
    type: PropTypes.string,
    approve: PropTypes.func,
    decline: PropTypes.func,
    message: PropTypes.object
};

export default ConfirmMessage;