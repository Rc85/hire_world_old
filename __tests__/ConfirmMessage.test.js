import React from 'react';
import Enzyme, { mount, render, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { User } from '../__mocks__/userMock';
import ConfirmMessage from '../src/components/includes/page/ConfirmMessage';
import toJson from 'enzyme-to-json';

const user = {user: User, status: 'get session success'};

Enzyme.configure({adapter: new Adapter()});
jest.setTimeout(30000);

describe('Confirm message in message details', () => {
    let jobProps = {};
    const wrapper = shallow(
        <ConfirmMessage
        prompt={true}
        user={user.user}
        message={{message_modified_date: new Date(), message_recipient: 'roger85', message_status: 'New'}}
        job={jobProps}
        decline={() => jest.fn()} />
    );

    describe('Snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Contains the appropriate sent and resent text', () => {
        expect(wrapper.find('small').at(0).text()).toContain('Sent');
        expect(wrapper.find('small').at(0).text()).toContain('Resent');
    });

    it('Should show approved message', () => {
        const wrapper = shallow(
            <ConfirmMessage
            prompt={true}
            user={user.user}
            message={{message_modified_date: new Date()}}
            job={{job_user_complete: true, job_client_complete: true}}
            decline={() => jest.fn()} />
        );

        expect(wrapper.find('#confirm-message-approved-message').text()).toEqual('You approved the request.');
    });

    it('Should show declined message', () => {
        const wrapper = shallow(
            <ConfirmMessage
            prompt={true}
            user={user.user}
            message={{message_modified_date: new Date()}}
            job={{job_user_complete: false, job_client_complete: false}}
            decline={() => jest.fn()} />
        );

        expect(wrapper.find('#confirm-message-declined-message').text()).toEqual('You declined the request.');
    });

    it('Should prompt for a reason', () => {
        wrapper.find('#confirm-message-decline-button').simulate('click');

        expect(wrapper.find('#confirm-message-reason-input')).toHaveLength(1);
        expect(wrapper.find('#confirm-message-reason-input').find('button').at(0).text()).toEqual('Submit');
    });

    it('Should show "new" indicator', () => {
        expect(wrapper.find('h4').text()).toEqual('New');
    });
});