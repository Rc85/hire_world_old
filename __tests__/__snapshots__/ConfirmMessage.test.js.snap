// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="alert alert-undefined mb-3 p-2"
  role="alert"
>
  <div
    className="d-flex-between-center"
  >
    <small
      className="text-muted"
    >
      Sent 
      a few seconds ago
       
      <span>
        (Resent 
        a few seconds ago
        )
      </span>
    </small>
    <h4>
      <span
        className="badge badge-success"
      >
        New
      </span>
    </h4>
  </div>
  <div
    className="text-center mb-3 keep-format"
  >
    <strong />
  </div>
  <hr />
  <div
    className="mb-3"
  >
    <button
      className="btn btn-success mr-1"
      id="confirm-message-approve-button"
      onClick={[Function]}
    >
      Approve
    </button>
    <button
      className="btn btn-danger"
      id="confirm-message-decline-button"
      onClick={[Function]}
    >
      Decline
    </button>
  </div>
  <div
    className="text-right"
  >
    <small
      className="text-muted"
    >
      Message ID: 
    </small>
  </div>
</div>
`;
